exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1333 @@
+	case kline.ThirtyMin:
+		return "30m"
+	case kline.OneHour:
+		return "1h"
+	case kline.TwoHour:
+		return "2h"
+	case kline.FourHour:
+		return "4h"
+	case kline.SixHour:
+		return "4h"
+	case kline.TwelveHour:
+		return "12h"
+	case kline.OneDay:
+		return "1d"
+	case kline.OneWeek:
+		return "1w"
+	case kline.OneMonth:
+		return "1M"
+	default:
+		return interval.Short()
+	}
+}
+
+// GetHistoricCandles returns candles between a time period for a set time interval
+func (by *Bybit) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if err := by.ValidateKline(pair, a, interval); err != nil {
+		return kline.Item{}, err
+	}
+
+	klineItem := kline.Item{
+		Exchange: by.Name,
+		Pair:     pair,
+		Asset:    a,
+		Interval: interval,
+	}
+	switch a {
+	case asset.Spot:
+		candles, err := by.GetKlines(ctx, pair.String(), by.FormatExchangeKlineInterval(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start, end)
+		if err != nil {
+			return klineItem, err
+		}
+
+		for x := range candles {
+			klineItem.Candles = append(klineItem.Candles, kline.Candle{
+				Time:   candles[x].StartTime,
+				Open:   candles[x].Open,
+				High:   candles[x].High,
+				Low:    candles[x].Low,
+				Close:  candles[x].Close,
+				Volume: candles[x].Volume,
+			})
+		}
+	case asset.CoinMarginedFutures, asset.Futures:
+		candles, err := by.GetFuturesKlineData(ctx, pair, by.FormatExchangeKlineInterval(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start)
+		if err != nil {
+			return klineItem, err
+		}
+
+		for x := range candles {
+			klineItem.Candles = append(klineItem.Candles, kline.Candle{
+				Time:   time.Unix(candles[x].OpenTime, 0),
+				Open:   candles[x].Open,
+				High:   candles[x].High,
+				Low:    candles[x].Low,
+				Close:  candles[x].Close,
+				Volume: candles[x].Volume,
+			})
+		}
+	case asset.USDTMarginedFutures:
+		candles, err := by.GetUSDTFuturesKlineData(ctx, pair, by.FormatExchangeKlineInterval(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start)
+		if err != nil {
+			return klineItem, err
+		}
+
+		for x := range candles {
+			klineItem.Candles = append(klineItem.Candles, kline.Candle{
+				Time:   time.Unix(candles[x].OpenTime, 0),
+				Open:   candles[x].Open,
+				High:   candles[x].High,
+				Low:    candles[x].Low,
+				Close:  candles[x].Close,
+				Volume: candles[x].Volume,
+			})
+		}
+
+	default:
+		return klineItem, fmt.Errorf("assetType not supported: %v", a)
+	}
+
+	klineItem.RemoveOutsideRange(start, end)
+	klineItem.SortCandlesByTimestamp(false)
+	return klineItem, nil
+}
+
+// GetHistoricCandlesExtended returns candles between a time period for a set time interval
+func (by *Bybit) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	return kline.Item{}, common.ErrNotYetImplemented
```

Comment:
Once Kline start date and date is added per the other nit, you can implement this too.